### PR TITLE
Differentiate in C/R between tokens with/out DB-challenge

### LINKED ISSUE #2491: is_challenge_response should not depent on the presence of a transaction_id in the database
The method ``is_challenge_response`` determines if the request is the response to a given challenge.
In this method it is also checked, if a challenge is available in the database table.
In certain scenarios this can lead to the effect, that privacyIDEA thinks, this request would not be a challenge.

We should remove the dependency on the availability of the challenge.

### PR DESCRIPTION
**Pull request by cornelinux**
A response to a challenge can arrive so late, that the
original challenge does not exist anymore in the database.
It still would be a response to a challenge, so we need to
differentiate between being a legit or non-legit response at all
(it has a "transaction_id")
and a response, for which still a challenge-DB-object exists.

When merging this PR, it will also rendet this PR useless:
Closes #2493

After all it...
Closes #2491

### CODE REVIEW

**Comment by cornelinux** [comment]
This is another approach than #2493. I like this here more, since it is clearer about categorizing the tokens in the ``token_lists`` in ``lib/token.py:check_token_list``.

File: privacyidea/lib/tokenclass.py [hunk 1]
```diff
@@ -1387,8 +1387,8 @@ def is_challenge_request(self, passw, user=None, options=None):
 
     def is_challenge_response(self, passw, user=None, options=None):
         """
-        This method checks, if this is a request, that is the response to
-        a previously sent challenge.
+        This method checks, if this is a request  that is supposed to be
```
Comment (plettich) [inline privacyidea/lib/tokenclass.py:1390]: please remove superfluous space

**Review by plettich — CHANGES_REQUESTED** [review]
just a minor typo

File: privacyidea/lib/tokenclass.py (hunk 1 above)
Comment (cornelinux) [reply privacyidea/lib/tokenclass.py:1390]: Hawk eye. Did not see that! Will do!

**Review by plettich — APPROVED** [review]
looks good

**Pull request merged**